tests/sentry/models/test_organizationmember.py

```diff
@@ -135,6 +136,30 @@ def test_delete_expired_clear(self):
         OrganizationMember.objects.delete_expired(timezone.now())
         assert OrganizationMember.objects.filter(id=member.id).first() is None
 
+    def test_delete_expired_SCIM_enabled(self):
+        organization = self.create_organization()
+        org2 = self.create_organization()
+        org3 = self.create_organization()
+        AuthProvider.objects.create(
+            provider="saml2", organization=organization, flags=AuthProvider.flags["scim_enabled"]
+        )
+        AuthProvider.objects.create(
+            provider="saml2", organization=org3, flags=AuthProvider.flags["allow_unlinked"]
+        )
+        AuthProvider.objects.create(
+            provider="okta", organization=org2, flags=AuthProvider.flags["scim_enabled"]
+        )
+        ninety_one_days = timezone.now() - timedelta(days=91)
+        member = OrganizationMember.objects.create(
+            organization=organization,
+            role="member",
+            email="test@example.com",
+            token="abc-def",
+            token_expires_at=ninety_one_days,
+        )
+        OrganizationMember.objects.delete_expired(timezone.now())
+        assert OrganizationMember.objects.filter(id=member.id).exists()
```

Comment:
Should we also validate the an org member associated with `org3` is successfully deleted?